### PR TITLE
Build JDK9 classes on JDK 9 and up

### LINKED ISSUE #25016: JDK 9+ classes only built for exactly JDK 9
With JDK 9 (now EOL:ed) I see entries like this:
```
[JDK9] Enabled [...-jdk9-only] directories to be compiled.
```

but with JDK 10 there are no such, I think the inclusion needs to be 9+ rather than exactly 9

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #25016

### CODE REVIEW

File: project/Jdk9CompileDirectoriesPlugin.scala [hunk 1]
```diff
@@ -9,20 +9,24 @@ import sbt.Keys._
 
 object Jdk9CompileDirectoriesPlugin extends AutoPlugin {
 
-  val jdkVersion: String = System.getProperty("java.version")
+  val MajorVersionRegex = """(\d+)(?:.+)?""".r
```
Comment (raboof) [inline project/Jdk9CompileDirectoriesPlugin.scala]: Is `"""(\d+).*"""` assuming too much?

**Review by raboof — APPROVED** [review]

**Comment by 2m** [comment]
Note that `Jdk9CompileDirectoriesPlugin` will be gone after merging the `build-everything-with-java-9` branch  is merged: https://github.com/akka/akka/issues/23789#issuecomment-385371766

**Comment by johanandren** [comment]
Ah, thanks for pointing that out @2m, didn't think of that.

**Comment by 2m** [comment]
Thanks for closing this. Less merge conflicts when the jdk9 is going to be merged.

**Pull request closed without merging**